### ISSUE TITLE
Would be nice to make it work with Abstract Type Members

### ISSUE THREAD

**Issue by cgarciaq** [Iterable/iterable-play-utils#9]
It does not work with abstract type members. The following code is not working, but would be very nice if it did:
```
trait Resource {
  type T <: Product
  
  def resourceForm = Form(CaseClassMapping.mapping[T])
  ...
}

trait ApplesComponent extends Resource {
  type T = Apple

  case class Apple(size:Int, color: String)
  ...
}
```
I think it can be achieved using WeakTypeTag, but I'm not sure.

**Comment by dontgitit** [comment]
Yeah it doesn't even work for type aliases: https://github.com/Iterable/iterable-play-utils/issues/2

You can try https://github.com/Iterable/play-formnolia which uses Magnolia under the hood, it might work with abstract types (though I haven't personally tried it for that use case)
